window_to_front/windows/window_to_front_plugin.cpp

```diff
@@ -0,0 +1,81 @@
+#include "include/window_to_front/window_to_front_plugin.h"
+
+// This must be included before many other Windows headers.
+#include <windows.h>
+
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar_windows.h>
+#include <flutter/standard_method_codec.h>
+
+#include <map>
+#include <memory>
+#include <sstream>
+
+namespace {
+
+class WindowToFrontPlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrarWindows *registrar);
+
+  WindowToFrontPlugin(flutter::PluginRegistrarWindows *registrar);
+
+  virtual ~WindowToFrontPlugin();
+
+ private:
+  // Called when a method is called on this plugin's channel from Dart.
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result);
+
+  // The registrar for this plugin, for accessing the window.
+  flutter::PluginRegistrarWindows *registrar_;
+};
+
+// static
+void WindowToFrontPlugin::RegisterWithRegistrar(
+    flutter::PluginRegistrarWindows *registrar) {
+  auto channel =
+      std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
+          registrar->messenger(), "window_to_front",
+          &flutter::StandardMethodCodec::GetInstance());
+
+  auto plugin = std::make_unique<WindowToFrontPlugin>(registrar);
+
+  channel->SetMethodCallHandler(
+      [plugin_pointer = plugin.get()](const auto &call, auto result) {
+        plugin_pointer->HandleMethodCall(call, std::move(result));
+      });
+
+  registrar->AddPlugin(std::move(plugin));
+}
+
+WindowToFrontPlugin::WindowToFrontPlugin(flutter::PluginRegistrarWindows *registrar) 
+  : registrar_(registrar) {}
+
+WindowToFrontPlugin::~WindowToFrontPlugin() {}
+
+void WindowToFrontPlugin::HandleMethodCall(
+    const flutter::MethodCall<flutter::EncodableValue> &method_call,
+    std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+  // Replace "getPlatformVersion" check with your plugin's method.
+  // See:
+  // https://github.com/flutter/engine/tree/master/shell/platform/common/cpp/client_wrapper/include/flutter
+  // and
+  // https://github.com/flutter/engine/tree/master/shell/platform/glfw/client_wrapper/include/flutter
+  // for the relevant Flutter APIs.
+  if (method_call.method_name().compare("activate") == 0) {
+    result->Success(flutter::EncodableValue(BringWindowToTop(
+      registrar_->GetView()->GetNativeWindow())));
```

Comment:
@stuartmorgan and @csells is there anything obviously wrong here? It compiles, but then doesn't bring the window to front. Sadly.

Reply:
are you calling [SetForegroundWindow](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setforegroundwindow)? I can't tell...

Reply:
Turns out https://stackoverflow.com/a/34414846/2142626 is the best answer I've found so far. And it's a doozy.